src/geo/map/cartodb-layer.js

```diff
@@ -125,11 +134,21 @@ var CartoDBLayer = LayerModelBase.extend({
   },
 
   update: function (attrs) {
```

Comment:
This method got a little tricky

Reply:
I think we can get rid of the Deprecated condition since it is implemented [in the builder](https://github.com/CartoDB/cartodb/blob/249b01951fa29d2a70d0a4efbdcc52eda9de1b81/lib/assets/core/javascripts/cartodb3/deep-insights-integration/layers-integration.js#L263-L272)

Reply:
I'll open a PR to simplify this.

Reply:
https://github.com/CartoDB/cartodb.js/pull/1823/files